jsonconfig generates many compiler warning

When running `./waf --checkall`, I see many `‘value’ may be used uninitialized in this function [-Wuninitialized]` warnings near `jubatus/core/common/jsonconfig/cast.hpp`. Why?

```
In file included from ../jubatus/core/unlearner/../common/jsonconfig.hpp:22:0,
                 from ../jubatus/core/unlearner/unlearner_factory.hpp:21,
                 from ../jubatus/core/unlearner/unlearner_factory.cpp:17:
/home/jubatus/Development/jubatus_core/jubatus/util/text/json/../../data/optional.h: In function ‘void jubatus::core::common::jsonconfig::serialize(jubatus::core::common::jsonconfig::json_config_iarchive_cast&, jubatus::util::data::serialization::named_value<jubatus::util::data::optional<T> >&) [with T = long int]’:
/home/jubatus/Development/jubatus_core/jubatus/util/text/json/../../data/optional.h:83:7: warning: ‘value’ may be used uninitialized in this function [-Wuninitialized]
../jubatus/core/unlearner/../common/./jsonconfig/cast.hpp:247:7: note: ‘value’ was declared here
```

--- a/jubatus/core/common/jsonconfig/cast.hpp
+++ b/jubatus/core/common/jsonconfig/cast.hpp
@@ -210,7 +210,7 @@ inline void serialize(json_config_iarchive_cast& js, std::map<K, V>& m) {
   typedef config::iterator iter_t;
   for (iter_t it = js.get_config().begin(), end = js.get_config().end();
       it != end; ++it) {
-    V value;
+    V value = V();
     json_from_config(it.value(), value, js.errors());
     tmp[it.key()] = value;
   }
@@ -229,7 +229,7 @@ inline void serialize(
   typedef config::iterator iter_t;
   for (iter_t it = js.get_config().begin(), end = js.get_config().end();
       it != end; ++it) {
-    V value;
+    V value = V();
     json_from_config(it.value(), value, js.errors());
     tmp[it.key()] = value;
   }
@@ -244,7 +244,7 @@ inline void serialize(
   using jubatus::util::text::json::json;
   if (js.get_config().contain(v.name)
       && (js.get_config()[v.name].get().type() != json::Null)) {
-    T value;
+    T value = T();
     json_from_config(js.get_config()[v.name], value, js.errors());
     v.v = value;
   } else {
@@ -294,14 +294,14 @@ void json_from_config(const config& conf, T& v, config_error_list* errors) {
 
 template <class T>
 T config_cast(const config& c) {
-  T value;
+  T value = T();
   json_from_config(c, value);
   return value;
 }
 
 template <class T>
 T config_cast(const config& c, config_error_list& errors) {
-  T value;
+  T value = T();
   json_config_iarchive_cast cast(c, &errors);
   serialize(cast, value);
   return value;
@@ -310,7 +310,7 @@ T config_cast(const config& c, config_error_list& errors) {
 template <class T>
 T config_cast_check(const config& c) {
   config_error_list errors;
-  T value;
+  T value = T();
   json_config_iarchive_cast cast(c, &errors);
   serialize(cast, value);
   if (!errors.empty()) {

